op-dispute-mon/Dockerfile

```diff
@@ -0,0 +1,17 @@
+ARG OP_STACK_GO_BUILDER=us-docker.pkg.dev/oplabs-tools-artifacts/images/op-stack-go:latest
```

Comment:
I think we need to update the `docker-bake.hcl` to have a `op-dispute-mon` target as well, similar to the `op-challenger` target it has. That will fix the `error: failed to find target op-dispute-mon` failure we're getting.